### DISCUSSION TITLE
some routes can be improved

### DISCUSSION THREAD

**Discussion by priyansh32** [NITK-KODE/cms-api#38] [General]
Currently there are two functions in `src/controllers/organizations` named `index` and `instituteIndex`

`index` fetched all organizations (I am not sure about the route of this function)  

`InstituteIndex` 
- fetches organizations belonging to a particular institute. 
- Route: `{{baseURL}}/institutes/:instituteId/organizations`

We can use a single route to get required organizations using query params.

**Comment by Ashley936** [comment]
I couldn't find the route for index function anywhere. I don't think it's even needed.

**Reply by priyansh32** [reply]
There isn't any right now, some routes need to be restructured.

**Comment by priyansh32** [comment]
I converted the issue into a discussion since some significant refactoring is required. Feel free to share ideas here

**Comment by Utkar5hM** [comment]
I have removed those specific routes like OrganizationIndex and InstitutionIndex but function for one of them was actually missing though(my mistake).
Now we can pass query for specific institute/organization/event _id instead of the instituteId or similar with others defined by us as the organizations,, which could be a problem but can be fixed easily. 

There needs to be a refactoring and fixing of the code especially with events and awards. especially with authorizations that are taking place in controllers and not in the authorization.js which is because a lot fo them had few changes among them due to which i could not refactor the code into one in a short time after adding multiple institutes support.. 

I am currently working on both the issues.

**Reply by Utkar5hM** [reply]
The first issue has been fixed. under #40

**Comment by priyansh32** [comment]
Some routes are currently like, `{{baseURL}}/institutes/:instituteID/organizations/:organizationId`, We are passing two unique Ids in URL  
Is it required? same for event routes.

**Reply by Utkar5hM** [reply]
I had initial intentions of making it this(independent routes) way, i did it this way first, but later had it changed for some reasons idr. If we want, this can be changed back to completely independent routes with queries instead.
